File: 백준/Silver/2503. 숫자 야구/숫자 야구.py
import sys

input = sys.stdin.readline

n = int(input())
arr = [list(map(int, input().split())) for _ in range(n)]
ans = 0

for a in range(1, 10):
    for b in range(1, 10):
        for c in range(1, 10):
            if a != b and a != c and b != c:
                cnt = 0
                for condi in arr:
                    num = condi[0]
                    strike = condi[1]
                    ball = condi[2]
                    s = 0
                    k = 0
                    p = num // 100
                    q = num % 100 // 10
                    r = num % 100 % 10
                    if a == p:
                        s += 1
                    elif a == q or a == r:
                        k += 1
                    if b == q:
                        s += 1
                    elif b == p or b == r:
                        k += 1
                    if c == r:
                        s += 1
                    elif c == p or c == q:
                        k += 1

                    if strike == s and ball == k:
                        cnt += 1
                    else:
                        break
                if cnt == n:
                    ans += 1
print(ans)

File: 백준/Silver/2503. 숫자 야구/숫자 야구.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));


        int n = Integer.parseInt(br.readLine());
        int[][] arr = new int[n][3];

        for (int i = 0; i < n; i++) {
            String[] temp = br.readLine().split(" ");
            arr[i][0] = Integer.parseInt(temp[0]);
            arr[i][1] = Integer.parseInt(temp[1]);
            arr[i][2] = Integer.parseInt(temp[2]);
        }

        int ans = 0;

        for (int i = 1; i <= 9; i++) {
            for (int j = 1; j <= 9; j++) {
                for (int k = 1; k <= 9; k++) {
                    if (i != j && i != k && j != k) {
                        boolean flag = true;

                        for (int[] temp : arr) {
                            int num = temp[0];
                            int strike = temp[1];
                            int ball = temp[2];

                            int a = num / 100;
                            int b = num % 100 / 10;
                            int c = num % 100 % 10;

                            int strikeNum = 0;
                            int ballNum = 0;

                            if (i == a) {
                                strikeNum++;
                            } else if (i == b || i == c) {
                                ballNum++;
                            }

                            if (j == b) {
                                strikeNum++;
                            } else if (j == c || j == a) {
                                ballNum++;
                            }
                            if (k == c) {
                                strikeNum++;
                            } else if (k == a || k == b) {
                                ballNum++;
                            }

                            if (strike != strikeNum || ball != ballNum) {
                                flag = false;
                                break;
                            }
                        }
                        if (flag) {
                            ans++;
                        }
                    }
                }
            }
        }
        System.out.println(ans);

    }
}